File: training-data/python/ini_parser.py
def parse_ini(text: str) -> dict:
    result = {}
    section = None
    for line in text.splitlines():
        line = line.strip()
        if not line or line.startswith(("#", ";")):
            continue
        if line.startswith("[") and line.endswith("]"):
            section = line[1:-1]
            result[section] = {}
        elif "=" in line and section is not None:
            key, _, value = line.partition("=")
            result[section][key.strip()] = value.strip()
    return result


if __name__ == "__main__":
    sample = """
    [server]
    host = localhost
    port = 8080
    """
    print(parse_ini(sample))

File: training-data/typescript/ini_parser.ts
type IniData = Record<string, Record<string, string>>;

function parseIni(text: string): IniData {
  const result: IniData = {};
  let section: string | null = null;
  for (let line of text.split('\n')) {
    line = line.trim();
    if (!line || line.startsWith('#') || line.startsWith(';')) continue;
    if (line.startsWith('[') && line.endsWith(']')) {
      section = line.slice(1, -1);
      result[section] = {};
    } else if (line.includes('=') && section !== null) {
      const [key, ...rest] = line.split('=');
      result[section][key.trim()] = rest.join('=').trim();
    }
  }
  return result;
}

const sample = `
[server]
host = localhost
port = 8080
`;
console.log(parseIni(sample));
